### PR TITLE
Config file: improve docs and help text

### PR DESCRIPTION
**Pull request by stsewd**
Close https://github.com/readthedocs/readthedocs.org/issues/8092

### CODE REVIEW

File: docs/config-file/v2.rst [hunk 1]
```diff
@@ -62,6 +62,11 @@ Read the Docs validates every configuration file.
 Any configuration option that isn't supported will make the build fail.
 This is to avoid typos and provide feedback on invalid configurations.
 
+.. note::
```
Comment (ericholscher) [inline docs/config-file/v2.rst:65]: These should probably be warnings?

File: readthedocs/templates/projects/project_advanced_settings_helptext.html [hunk 2]
```diff
@@ -4,7 +4,6 @@
     These settings can be configured using a
     <a href="{{ config_file_link }}">configuration file</a>.
     That's the recommended way to set up your project.
-    Settings in the configuration
-    file override the settings listed here.
+    Settings listed here are ignored when using a configuration file.
```
Comment (ericholscher) [inline readthedocs/templates/projects/project_advanced_settings_helptext.html:7]: This should probably be bold.

**Review by ericholscher — APPROVED** [review]

**Review by astrojuanlu — COMMENTED** [review]
I agree with both comments @ericholscher made

**Comment by stsewd** [comment]
Changed :+1:

**Pull request merged**